packages/react-components/src/components/CaptionsSettingsModal.tsx

```diff
@@ -103,11 +103,11 @@ export const _CaptionsSettingsModal = (props: _CaptionsSettingsModalProps): JSX.
     key:
       currentCaptionLanguage !== ''
         ? currentCaptionLanguage
-        : _spokenLanguageToCaptionLanguage[selectedSpokenLanguage.key],
+        : _spokenLanguageToCaptionLanguage[selectedSpokenLanguage.key as keyof SpokenLanguageStrings],
```

Comment:
We should also be unfortunately avoiding `as` as much as possible, it has a similar drawback to implicit any where the casting might not be correct.
E.g., `selectedSpokenLanguage.key` is type string and there is no compile-time type safety to determinately say this is of type `keyof SpokenLanguageStrings`.

For this instance would declaring something like this work instead:
```
type LanguageDropdownOption = IDropdownOption<SpokenLanguageStrings> & {
  key: keyof SpokenLanguageStrings;
};
```
then `const [selectedSpokenLanguage, setSelectedSpokenLanguage] = useState<IDropdownOption>` becomes
```
const [selectedSpokenLanguage, setSelectedSpokenLanguage] = useState<LanguageDropdownOption>
```

Quickly looking, you may have to also strongly type the _CaptionsSettingsModalProps, instead of being `string`, they should be
```
/** @internal */
export type _SupportedLanguage = keyof SpokenLanguageStrings;

...
export interface _CaptionsSettingsModalProps {
  supportedSpokenLanguages: _SupportedLanguage[];
  onSetSpokenLanguage: (language: _SupportedLanguage) => Promise<void>;
  currentSpokenLanguage: _SupportedLanguage;
  etc.
```

Reply:
This method causes some issues up the chain, more specifically on the onChange event from IDropdown.
![image](https://github.com/Azure/communication-ui-library/assets/97130533/c58a861c-1df5-45fc-a7ad-1e5171ca77c1)


Have any particular suggestions on how to "override" the string requirement from IDropdownOptions?

Reply:
Hmm, because the `<Dropdown />` component can't take in a generic `<T>`, it can't pass the types of the keys supplied back into the onChange event and correlate those two types. That's a place we'll have to have an unsafe "as" cast.

(Technically we could write a higher-order component that encapsulates the unsafe typing with a guard that proves the type safety by extending the existing fluent component, but that's overkill.)

Instead that function will have to be something like:
```ts
onChange={(ev, option) => onSpokenLanguageChange(ev, option as LanguageDropdownOption)}
```